### DISCUSSION TITLE
suggestion for app/api.    Prevent discharge to make battery space.

### DISCUSSION THREAD

**Discussion by rob0101** [amberelectric/public-api#152] [Ideas]
Today and yesterday Smartshift saw a good negative price in the near future forecast and started to sell from the battery to make room to buy at the cheap price.  I can see the logic and how it might suit some customers, but...

can we have a way to totally disable that action ?

My automations add loads to take advantage of negative prices when they happen. Discharging the battery to make space can never match the returns from my automations just adding more loads. At its worst (a failed punt) it wastes both money and battery cycles for everyone.

**Comment by madpilot** [comment]
Yeah, we have plans for this. Two actually:

1) the ability to say don't do that, and
2) a way to inform us of your load curve so we can take that into account

Hoping to get that in play next quarter.

**Reply by rob0101** [reply]
Can’t wait for all the new options.

**Comment by purcell-lab** [comment]
> can we have a way to totally disable that action ?
> 
> My automations add loads to take advantage of negative prices when they happen.

In this situation I have my automations set my backup reserve % to match my current SOC which prevents SmartShift from exporting.

With powerwall you also have the option to limit exports to solar/everything which also prevents battery discharge to the grid.

![Screenshot_20230805-080803](https://github.com/amberelectric/public-api/assets/79175134/49c16f33-a8dd-42d2-bba7-6645fe2b4391)

Looking forward to being able to inject my load forecast which will improve SmartShift calculations.
